src/ensembl/src/content/app/browser/state/drawer/types.ts

```diff
@@ -14,27 +14,33 @@
  * limitations under the License.
  */
 
-export enum DrawerView {
-  BOOKMARKS = 'bookmarks',
-  TRACK_DETAILS = 'track_details',
-  GENE_SUMMARY = 'gene_summary',
-  TRANSCRIPT_SUMMARY = 'transcript_summary'
-}
+type BookmarksDrawerView = {
+  name: 'bookmarks';
+};
 
-export type DrawerStateForGenome = Readonly<{
-  isDrawerOpened: boolean;
-  activeDrawerView: DrawerView | null;
-  activeDrawerTrackId: string | null;
-  activeDrawerTranscriptId: string | null;
-}>;
+export type GeneDrawerView = {
+  name: 'gene_summary';
+  payload: {
+    geneId: string; // in ensObjectId format
+  };
+};
 
-export type DrawerState = Readonly<{
-  [genomeId: string]: DrawerStateForGenome;
-}>;
+export type TranscriptDrawerView = {
+  name: 'transcript_summary';
+  payload: {
```

Comment:
Why do we need `payload` instead of having IDs at the same level as name?

Reply:
I was imitating redux actions.

Reply:
Agreed to move the contents of the payload field to the top level; done.